modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/config/BeanCreationFailureAnalyzer.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.console.config;
+
+import org.apache.ignite.internal.util.typedef.X;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.boot.diagnostics.AbstractFailureAnalyzer;
+import org.springframework.boot.diagnostics.FailureAnalysis;
+import org.springframework.boot.diagnostics.analyzer.AbstractInjectionFailureAnalyzer;
+
+/**
+ * An {@link AbstractInjectionFailureAnalyzer} that performs analysis of failures caused
```

Comment:
Looks like copy-paste.
Lets use "Custom failures analyzer that will report failure reason without polluting logs."

Reply:
He don't report failure reason, just extract error. 
Look at javadoc of FailureAnalyzer or it implementation in spring framework.
I don't think that we should describe something more.